runtimes/runtimes/lsp/router/lspServer.ts

```
@@ -87,22 +87,29 @@ export class LspServer {
         params: InitializeParams,
         token: CancellationToken
     ): Promise<PartialInitializeResult | ResponseError<InitializeError> | undefined> => {
-        this.clientSupportsNotifications =
-            params.initializationOptions?.aws.awsClientCapabilities?.window?.notifications
+        try {
+            this.clientSupportsNotifications =
+                params.initializationOptions?.aws?.awsClientCapabilities?.window?.notifications
 
-        if (!this.initializeHandler) {
-            return
-        }
+            if (!this.initializeHandler) {
+                return
+            }
 
-        const initializeResult = await asPromise(this.initializeHandler(params, token))
-        if (!(initializeResult instanceof ResponseError)) {
-            this.initializeResult = initializeResult
-            if (initializeResult?.serverInfo) {
-                this.notificationRouter = new RouterByServerName(initializeResult.serverInfo.name, this.encoding)
+            const initializeResult = await asPromise(this.initializeHandler(params, token))
+            if (!(initializeResult instanceof ResponseError)) {
+                this.initializeResult = initializeResult
+                if (initializeResult?.serverInfo) {
+                    this.notificationRouter = new RouterByServerName(initializeResult.serverInfo.name, this.encoding)
+                }
             }
-        }
 
-        return initializeResult
+            return initializeResult
+        } catch (error) {
+            this.lspConnection.console.log(
+                `Error in initialize handler: "${error}",\nwith initialization options: ${JSON.stringify(params.initializationOptions)}`
+            )
+            return
```

Would we still want to throw the error? This would inform clients that the server cannot start up. (We would want to return the formatted message above as the error state message). 

To put it another way: Is there a chance that by swallowing the error, the server will be in an unstable state (with unpredictable behavior)?

---

fix in this PR: https://github.com/aws/language-server-runtimes/pull/307